tests/integration_tests/datasources/test_azure.py

```diff
@@ -34,6 +37,15 @@ def test_azure_eject(session_cloud: IntegrationCloud):
     ) as instance:
         source = get_validated_source(session_cloud)
         if source.installs_new_version():
+            if (
```

Comment:
Why only in this test? Wouldn't we need to do an update on any test being launched on noble that installs something new? I would think this should go into `install_new_cloud_init`or `install_deb` directly.

Reply:
That's a better place, I had thought about it but considered that this should only affect MINIMAL images in general or days in which we've changed strict packaging dependencies in an additive fashion and haven't released those pkg changes to cloud-init yet. I had also thought to avoid the cost of the apt update... but it's only a cost paid at image setup time so that's ok. I'll move it.